geniza/corpus/management/commands/export_metadata.py

```diff
@@ -1,30 +1,189 @@
+class MetadataExportRepo(Timerable):
+    local_path_key = "METADATA_REPO_LOCAL_PATH"
+    remote_url_key = "METADATA_REPO_REMOTE_URL"
+
+    repo_dir_data = "data"
+    ext_csv = ".csv"
+
+    def __init__(
+        self, local_path=None, remote_url=None, print_func=None, progress=True
+    ):
+        self._local_path = local_path
+        self._remote_url = remote_url
+        self.print = print_func if print_func is not None else print
+        self.progress = progress
+
+        # make sure repo exists and is initialized in directory
+        try:
+            if not os.path.exists(self.local_path):
+                with self.timer("Cloning repository"):
+                    Repo.clone_from(url=self.remote_url, to_path=self.local_path)
+
+            # set repo obj
+            self.repo = Repo(self.local_path)
+        except GitCommandError as e:
+            raise CommandError(
+                f"Git command failed. Does repository exist?\n\nError message:\n\n{e}"
+            )
+
+    @cached_property
+    def local_path(self):
+        if not (self._local_path or hasattr(settings, self.local_path_key)):
+            raise CommandError(
+                f"Please set {self.local_path_key} in local_settings.py or settings file."
+            )
+        return os.path.abspath(
+            self._local_path or getattr(settings, self.local_path_key)
+        )
+
+    @cached_property
+    def remote_url(self):
+        if not (self._remote_url or hasattr(settings, self.remote_url_key)):
+            raise CommandError(
+                f"Please set {self.remote_url_key} in local_settings.py or settings file."
+            )
+        return self._remote_url or getattr(settings, self.remote_url_key)
+
+    @cached_property
+    def path_data(self):
+        odir = os.path.join(self.local_path, self.repo_dir_data)
+        if not os.path.exists(odir):
+            os.makedirs(odir)
+        return odir
+
+    def get_path_csv(self, docname):
+        return os.path.join(self.path_data, docname + self.ext_csv)
+
+    @cached_property
+    def path_documents_csv(self):
+        return self.get_path_csv("documents")
+
+    @cached_property
+    def path_footnotes_csv(self):
+        return self.get_path_csv("footnotes")
+
+    @cached_property
+    def path_sources_csv(self):
+        return self.get_path_csv("sources")
+
+    @cached_property
+    def path_fragments_csv(self):
+        return self.get_path_csv("fragments")
+
+    @cached_property
+    def paths(self):
+        return [
+            self.path_documents_csv,
+            self.path_footnotes_csv,
+            self.path_sources_csv,
+            self.path_fragments_csv,
+        ]
+
+    def add_and_commit(self):
+        for fn in self.paths:
+            if os.path.exists(fn):
+                self.repo.index.add(fn)
+        self.repo.index.commit(
+            "Auto-syncing @ " + timezone.now().strftime("%Y-%m-%d %H:%M:%S")
+        )
+
+    def write_local(self):
```

Comment:
This name seems kind of opaque to me. generate_csvs ? export_csvs ? generate_exports ? (maybe not csv-specific if we still hope to add json at some point)